tests/unit/src/test/scala/tests/codeactions/ImportMissingSymbolLspSuite.scala

```diff
@@ -78,26 +78,102 @@ class ImportMissingSymbolLspSuite
-    s"""|${ImportMissingSymbol.title("Instant", "java.time")}
+    s"""|${ImportMissingSymbol.allSymbolsTitle}
+        |${ImportMissingSymbol.title("Instant", "java.time")}
+        |${ImportMissingSymbol.title("ListBuffer", "scala.collection.mutable")}
+        |${CreateNewSymbol.title("Instant")}
+        |${CreateNewSymbol.title("ListBuffer")}
+        |""".stripMargin,
+    """|package a
+       |
+       |import scala.concurrent.Future
+       |import java.time.Instant
+       |import scala.collection.mutable
+       |
+       |object A {
+       |  val f = Future.successful(Instant.now)
+       |  val b = mutable.ListBuffer.newBuilder[Int]
+       |}
+       |""".stripMargin,
+    expectNoDiagnostics = false
+  )
+
+  check(
+    "multi-across-lines-ambiguous-and-non-ambiguous",
+    """|package a
+       |
+       |object A {
+       |  val f = <<Future.successful(Instant.now)
+       |  val b = ListBuffer.newBuilder[Int]>>
+       |}
+       |""".stripMargin,
+    s"""|${ImportMissingSymbol.allSymbolsTitle}
+        |${ImportMissingSymbol.title("Future", "scala.concurrent")}
+        |${ImportMissingSymbol.title("Future", "java.util.concurrent")}
+        |${ImportMissingSymbol.title("Instant", "java.time")}
         |${ImportMissingSymbol.title("ListBuffer", "scala.collection.mutable")}
+        |${CreateNewSymbol.title("Future")}
         |${CreateNewSymbol.title("Instant")}
         |${CreateNewSymbol.title("ListBuffer")}
         |""".stripMargin,
     """|package a
        |
        |import java.time.Instant
        |
+       |import scala.collection.mutable
+       |
        |object A {
        |  val f = Future.successful(Instant.now)
+       |  val b = mutable.ListBuffer.newBuilder[Int]
+       |}
+       |""".stripMargin,
+    expectNoDiagnostics = false
+  )
+
+  check(
+    "multi-same-symbol",
+    """|package a
+       |
+       |object A {
+       |  val f = <<Future.successful(Instant.now)
        |  val b = ListBuffer.newBuilder[Int]
+       |  val t = Future.successful(ListBuffer.empty)>>
+       |}
+       |""".stripMargin,
+    s"""|${ImportMissingSymbol.allSymbolsTitle}
+        |${ImportMissingSymbol.title("Future", "scala.concurrent")}
+        |${ImportMissingSymbol.title("Future", "java.util.concurrent")}
+        |${ImportMissingSymbol.title("Instant", "java.time")}
+        |${ImportMissingSymbol.title("ListBuffer", "scala.collection.mutable")}
+        |${ImportMissingSymbol.title("Future", "scala.concurrent")}
+        |${ImportMissingSymbol.title("Future", "java.util.concurrent")}
+        |${ImportMissingSymbol.title("ListBuffer", "scala.collection.mutable")}
+        |${CreateNewSymbol.title("Future")}
+        |${CreateNewSymbol.title("Instant")}
+        |${CreateNewSymbol.title("ListBuffer")}
+        |${CreateNewSymbol.title("Future")}
+        |${CreateNewSymbol.title("ListBuffer")}
+        |""".stripMargin,
+    """|package a
+       |
+       |import java.time.Instant
+       |
```

Comment:
I don't know why there's a space here ... Moreover tests sometimes pass and sometimes not cause the imports are reversed

Reply:
If you add a sort in the code action that returns them, does it fix this?

Reply:
I wanted to keep the "import all" above all the others but it does not work quite well anyway.. I'll try different ways

Reply:
It will not fix anyway because it's not the order of actions but the order of TextEdits that could matter.
But actually it looks like it's kind of random, text edits order is always the same but they seems not to be applied in that order.

=> after some checks I found that it's only on tests. On files when I use the codeAction they are always applied in the same order. (Could be the textEdits.scala file)

Reply:
About the spaces it's also a textEdits issue. If a file has no imports yet, imports textEdits are prepend with a \n. In our case we concatenate textEdits so they all are prepend with it..

I'm not sure I can fix this two issues

Reply:
Ahhhh of course...
https://github.com/scalameta/metals/pull/1051
I should have known that.. seeing that I added it 😬 .

I suppose this is an edge case that you'll only hit on if you are triggering this code action from a file that has no imports yet. 🤔 I'm trying to think what the best approach to fixing this is. We could just check the text edits and strip out the leading `\n` on the second textEdit, but that will still fail in this test if it's flaky. I'm not 100% sure why the textEdits aren't applied in a uniform way in tests. @tgodzik may know more on that.

Reply:
We do sort them by start to apply them properly:
```
      val positions = edits
        .map(edit => edit -> edit.getRange.toMeta(input))
        .sortBy(_._2.start)
```
in `TextEdits.scala`. I am not sure what will be the behaviour in case of multiple imports, but I believe we need to fix it there.